src/Umbraco.Infrastructure/PropertyEditors/TagsPropertyEditor.cs

```diff
@@ -146,46 +167,45 @@ public IEnumerable<ITag> GetTags(object? value, object? dataTypeConfiguration, i
                     throw new NotSupportedException($"Value \"{tagConfiguration.StorageType}\" is not a valid TagsStorageType.");
             }
 
-            return tags.Select(x => new Tag
-            {
-                Group = tagConfiguration.Group,
-                Text = x,
-                LanguageId = languageId,
-            });
+            return tags.ToArray();
         }
 
-
         /// <inheritdoc />
         public override IValueRequiredValidator RequiredValidator => new RequiredJsonValueValidator();
 
-        /// <inheritdoc />
-        public override object? FromEditor(ContentPropertyData editorValue, object? currentValue)
+        public override object? ToEditor(IProperty property, string? culture = null, string? segment = null)
         {
-            var value = editorValue.Value?.ToString();
-
-            if (string.IsNullOrEmpty(value))
+            var val = property.GetValue(culture, segment);
+            if (val == null)
             {
                 return null;
             }
 
-            var tagConfiguration = editorValue.DataTypeConfiguration as TagConfiguration ?? new TagConfiguration();
-            if (tagConfiguration.Delimiter == default)
-                tagConfiguration.Delimiter = ',';
+            IDataType? dataType = _dataTypeService.GetDataType(property.PropertyType.DataTypeId);
+            TagConfiguration configuration = dataType?.ConfigurationObject as TagConfiguration ?? new TagConfiguration();
+            var tags = ParseTags(val, configuration);
 
-            string[] trimmedTags = Array.Empty<string>();
+            return tags.Any() ? tags : null;
+        }
 
-            if (editorValue.Value is JArray json)
+        /// <inheritdoc />
+        public override object? FromEditor(ContentPropertyData editorValue, object? currentValue)
+        {
+            if (editorValue.Value is not IEnumerable<string> stringValues)
             {
-                trimmedTags = json.HasValues ? json.Select(x => x.Value<string>()).OfType<string>().ToArray() : Array.Empty<string>();
+                return null;
             }
-            else if (string.IsNullOrWhiteSpace(value) == false)
+
+            var trimmedTags = stringValues.Select(s => s.Trim()).Where(s => s.IsNullOrWhiteSpace() == false).ToArray();
+            if (trimmedTags.Any() == false)
```

Comment:
More pattern matching 😁 
```suggestion
            if (trimmedTags.Any() is false)
```